internal/scenes/game/systems/menu.go

```diff
@@ -0,0 +1,173 @@
+		// the container will use a plain color as its background
+		widget.ContainerOpts.BackgroundImage(image.NewNineSliceColor(color.NRGBA{0, 0, 0, 175})),
+
+		// the container will use an anchor layout to layout its single child widget
+		widget.ContainerOpts.Layout(widget.NewAnchorLayout()),
+	)
+
+	menuContainer := widget.NewContainer(
+		widget.ContainerOpts.WidgetOpts(
+			widget.WidgetOpts.LayoutData(widget.AnchorLayoutData{
+				HorizontalPosition: widget.AnchorLayoutPositionCenter,
+				VerticalPosition:   widget.AnchorLayoutPositionCenter,
+			}),
+		),
+		widget.ContainerOpts.Layout(
+			widget.NewRowLayout(
+				widget.RowLayoutOpts.Direction(widget.DirectionVertical),
+				widget.RowLayoutOpts.Spacing(30),
+			),
+		),
+	)
+
+	newGameButton := createMenuButton("New Game", func(args *widget.ButtonClickedEventArgs) {
+		println("options button clicked")
+	})
+	optionsButton := createMenuButton("Options", func(args *widget.ButtonClickedEventArgs) {
+		println("options button clicked")
+	})
+	quitButton := createMenuButton("Quit", func(args *widget.ButtonClickedEventArgs) {
+		os.Exit(1)
+	})
+
+	// add the newGameButton as a child of the container
+	menuContainer.AddChild(newGameButton)
+	menuContainer.AddChild(optionsButton)
+	menuContainer.AddChild(quitButton)
+	rootContainer.AddChild(menuContainer)
+
+	return rootContainer
+}
+
+func createMenuButton(buttonText string, handler func(args *widget.ButtonClickedEventArgs)) *widget.Button {
+	buttonImage, _ := loadButtonImage()
+	face, _ := loadFont(80)
+	// construct a newGameButton
+	button := widget.NewButton(
+		// set general widget options
+		widget.ButtonOpts.WidgetOpts(
+			// instruct the container's anchor layout to center the newGameButton both horizontally and vertically
+			widget.WidgetOpts.LayoutData(widget.RowLayoutData{
+				Position: widget.RowLayoutPositionCenter,
+			}),
+		),
+
+		// specify the images to use
+		widget.ButtonOpts.Image(buttonImage),
+
+		// specify the newGameButton's text, the font face, and the color
+		widget.ButtonOpts.Text(buttonText, face, &widget.ButtonTextColor{
+			Idle: color.NRGBA{0xdf, 0xf4, 0xff, 0xff},
+		}),
+
+		// specify that the newGameButton's text needs some padding for correct display
+		widget.ButtonOpts.TextPadding(widget.Insets{
+			Left:   30,
+			Right:  30,
+			Top:    5,
+			Bottom: 5,
+		}),
+
+		// add a handler that reacts to clicking the newGameButton
+		widget.ButtonOpts.ClickedHandler(handler),
+	)
+	return button
+}
+
+func (system *menusystem) Update(e *ecs.ECS) {
+	// ui.Update() must be called in ebiten Update function, to handle user input and other things
+	var ok bool
+	if system.menuEntry == nil {
+		if system.menuEntry, ok = components.Menu.First(e.World); !ok {
+			panic("no menu found")
+		}
+	}
+
+	menu := components.Menu.Get(system.menuEntry)
+
+	if inpututil.IsKeyJustPressed(ebiten.KeyEscape) {
+		log.Print(menu.IsOpen)
+
+		menu.ToggleMenu()
+
+		//Update function doesn't seem to respect IsPaused yet
```

Comment:
rest of the game keeps moving regardless

Reply:
I think we need to handle that ourselves, by not drawing/updating specific systems when `IsPaused` is true. `pause()` and `resume()` only stop/resume the internal timer, but the systems are still processed.